pkg/cmd/server/api/types.go

```diff
@@ -324,6 +328,31 @@ type MasterConfig struct {
 	AuditConfig AuditConfig
 }
 
+// MasterAuthConfig configures authentication options in addition to the standard
+// oauth token and client certificate authenticators
+type MasterAuthConfig struct {
+	RequestHeader *RequestHeaderAuthenticationOptions
+}
+
+// RequestHeaderAuthenticationOptions provides options for setting up a front proxy against the entire
+// API instead of against the /oauth endpoint.
+type RequestHeaderAuthenticationOptions struct {
+	// ClientCA is a file with the trusted signer certs.  If empty, no request verification is done, and any direct request to the OAuth server can impersonate any identity from this provider, merely by setting a request header.
+	ClientCA string
```

Comment:
> Needs to make it into the client ca pool used by the server

Yes, one step at a time I think.  I'll open an issue, but I think the wiring was nasty until 1.6.